bears/xml/XMLBear.py

```diff
@@ -1,10 +1,27 @@
 from coalib.bearlib.abstractions.Lint import Lint
 from coalib.bears.LocalBear import LocalBear
+from coalib.misc.Shell import escape_path_argument
+from coalib.settings.Setting import path
+from coalib.settings.Setting import url
+
+
+def path_or_url():
+    '''
+    Creates a function which converts the given argument into a ``url`` or
+    ``path``.
+
+    :return: Returns a conversion function.
```

Comment:
Why do we need a decorated function here ?

just do:
```
    def path_or_url(xml_dtd):
        try:
            return url(xml_dtd)
        except ValueError:
            return path(xml_dtd)
```
and then use:
`xml_dtd: path_or_url=""`